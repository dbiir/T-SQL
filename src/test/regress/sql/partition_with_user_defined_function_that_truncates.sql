-- Given there is a partitioned table
	create table some_partitioned_table_to_truncate
	(
		a integer
	)
	partition by range (a) (
		partition b start (0)
	);

-- And a function that truncates the partitioned table
	CREATE OR REPLACE FUNCTION truncate_a_partition_table() RETURNS VOID AS
	$$
	BEGIN
	    TRUNCATE some_partitioned_table_to_truncate;
	END;
	$$ LANGUAGE plpgsql;

-- When I truncate the table twice
	insert into some_partitioned_table_to_truncate
	       select i from generate_series(1, 10) i;
   	select count(*) from some_partitioned_table_to_truncate;
	select truncate_a_partition_table();
   	select count(*) from some_partitioned_table_to_truncate;	

-- Then I get the same result both times (no rows)
	insert into some_partitioned_table_to_truncate
	       select i from generate_series(1, 10) i;
   	select count(*) from some_partitioned_table_to_truncate;
	select truncate_a_partition_table();
   	select count(*) from some_partitioned_table_to_truncate;	

