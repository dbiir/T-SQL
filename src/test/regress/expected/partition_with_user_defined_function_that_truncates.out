-- Given there is a partitioned table
	create table some_partitioned_table_to_truncate
	(
		a integer
	)
	partition by range (a) (
		partition b start (0)
	);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'a' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
NOTICE:  CREATE TABLE will create partition "some_partitioned_table_to_truncate_1_prt_b" for table "some_partitioned_table_to_truncate"
-- And a function that truncates the partitioned table
	CREATE OR REPLACE FUNCTION truncate_a_partition_table() RETURNS VOID AS
	$$
	BEGIN
	    TRUNCATE some_partitioned_table_to_truncate;
	END;
	$$ LANGUAGE plpgsql;
-- When I truncate the table twice
	insert into some_partitioned_table_to_truncate
	       select i from generate_series(1, 10) i;
   	select count(*) from some_partitioned_table_to_truncate;
 count 
-------
    10
(1 row)

	select truncate_a_partition_table();
 truncate_a_partition_table 
----------------------------
 
(1 row)

   	select count(*) from some_partitioned_table_to_truncate;	
 count 
-------
     0
(1 row)

-- Then I get the same result both times (no rows)
	insert into some_partitioned_table_to_truncate
	       select i from generate_series(1, 10) i;
   	select count(*) from some_partitioned_table_to_truncate;
 count 
-------
    10
(1 row)

	select truncate_a_partition_table();
 truncate_a_partition_table 
----------------------------
 
(1 row)

   	select count(*) from some_partitioned_table_to_truncate;	
 count 
-------
     0
(1 row)

